pkg/cli/serve/serve.go

```diff
@@ -69,23 +85,53 @@ func NewServeCmd() *cobra.Command {
 		Long:    serviceLong,
 		Example: serviceExample,
 		Args:    cobra.OnlyValidArgs,
+		PreRun: func(cmd *cobra.Command, args []string) {
+			clientID, _ := cmd.Flags().GetString(config.OCMClientID)
+			clientSecret, _ := cmd.Flags().GetString(config.OCMClientSecret)
+			mode, _ := cmd.Flags().GetBool(config.FleetMode)
+
+			// Mark AccessToken and ClusterID as required only in default mode
+			if !mode && clientID == "" && clientSecret == "" {
```

Comment:
This means we will need to set the flags only when both the values are empty?

Reply:
Yes. It's not a nice solution but want I intended to do is that:
- If running `ocm-agent` in default mode, the only acceptable flags should be `--access-token` and `--cluster-id`.
- If running `ocm-agent` in fleet mode (`--fleet-mode`), the `--ocm-client-id` and `--ocm-client-secret` are optional flags for testing but are only valid in fleet mode. 
- For either of the modes, added validations that there shouldn't be a mix of the flags while testing and if it is then the CLI should report the expected behaviour.